packages/client/hmi-client/src/components/widgets/tera-common-modal-dialogs.vue

```diff
@@ -45,7 +63,21 @@ const currentRoute = useCurrentRoute();
 
 // For now, we just use project-menu.ts to manage modals related to projects
 // For non-project related modals we may want to create new composable or abstract project-menu.ts into a modal manager
-const { isShareDialogVisible, isRemoveDialogVisible, isProjectConfigDialogVisible, menuProject } = useProjectMenu();
+const { isCopyDialogVisible, isShareDialogVisible, isRemoveDialogVisible, isProjectConfigDialogVisible, menuProject } =
+	useProjectMenu();
+
+const projectName = ref<string>('');
+const project = ref();
+
+const saveCopy = async () => {
+	isCopyDialogVisible.value = false;
+	router.push(RoutePath.Home);
+
+	const copiedProject = await useProjects().clone(project.value.id);
+	if (!copiedProject) return;
+	copiedProject.name = projectName.value;
+	useProjects().update(copiedProject);
+};
```

Comment:
Your have a router.push _before_  you started to copy, would this create a problem if the project is large (decent amount of workflows, datasets, documents models ... etc) and the clone functions takes more than a few seconds?

Reply:
I had no issues copying the past monthly demo